### PR TITLE
OCPBUGS-34005: openstack: Attach a bootstrap FIP to the cluster network port

### PR DESCRIPTION
**Pull request by pierreprinetti**
Before this patch, the bootstrap FIP was attached to the first available port attached to the bootstrap server. As a consequence, the Installer sometimes tried to attach the FIP to a port that was created on an additional network.

With this patch, the bootstrap FIP is only attached to the bootstrap port that exists on the cluster network.

### CODE REVIEW

**Review by mdbooth — COMMENTED** [review]
/lgtm

Incidentally this is what CAPO does: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/1c78326d8aad3fb229d7feec6a1888df20386f77/pkg/cloud/services/networking/port.go#L75-L125

However, CAPO can't make the same assumptions as we can make here, so this is fine.

File: pkg/infrastructure/openstack/postprovision/floatingips.go [hunk 1]
```diff
@@ -48,27 +49,28 @@ func FloatingIPs(ctx context.Context, c client.Client, cluster *capo.OpenStackCl
 	return nil
 }
 
-// Get the first port associated with an instance.
+// Get the first port associated with an instance existing on the given port.
```
Comment (mandre) [inline pkg/infrastructure/openstack/postprovision/floatingips.go:52]:
```suggestion
// Get the first port associated with an instance existing on the given network.
```

**Review by mandre — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**